src/github.com/t3chguy/riot-static/templates.go

```diff
@@ -0,0 +1,166 @@
+	},
+	"plus": func(a, b int) int {
+		return a + b
+	},
+	"minus": func(a, b int) int {
+		return a - b
+	},
+	"URL": func(str string) template.URL {
+		return template.URL(str)
+	},
+	"HTML": func(str string) template.HTML {
+		return template.HTML(str)
+	},
+	"mRoomMember": func(event *gomatrix.Event) interface{} {
+		// join -> join = avatar/display name
+		// join -> quit = kick/leave
+		// * -> join = join
+		// * -> invite = invite
+		var content, prevContent *MemberEventContent
+
+		dataContent, _ := json.Marshal(event.Content)
+		if err := json.Unmarshal(dataContent, &content); err != nil {
+			content = &MemberEventContent{}
+		}
+
+		var effect string
+
+		dataPrevContent, _ := json.Marshal(event.PrevContent)
+		if err := json.Unmarshal(dataPrevContent, &prevContent); err != nil {
+			prevContent = &MemberEventContent{}
+		}
+
+		sender := event.Sender
+		target := *event.StateKey
+
+		switch content.Membership {
+		case "invite":
+			return sender + " invited " + target + "."
+		case "ban":
+			return sender + " banned " + target + "(" + event.Content["reason"].(string) + ")."
+		case "join":
+			if event.PrevContent != nil && prevContent.Membership == "join" {
+				if prevContent.DisplayName == "" && content.DisplayName != "" {
+					return sender + " set their display name to " + content.DisplayName + "."
+				} else if prevContent.DisplayName != "" && content.DisplayName == "" {
+					return sender + " removed their display name " + prevContent.DisplayName + "."
+				} else if prevContent.DisplayName != content.DisplayName {
+					return sender + " changed their display name from " + prevContent.DisplayName + " to " + content.DisplayName + "."
+				} else if prevContent.AvatarURL == "" && content.AvatarURL != "" {
+					return sender + " set a profile picture."
+				} else if prevContent.AvatarURL != "" && content.AvatarURL == "" {
+					return sender + " removed their profile picture."
+				} else if prevContent.AvatarURL != content.AvatarURL {
+					return sender + " changed their profile picture."
+				} else {
+					return ""
+				}
+			} else {
+				return target + " joined the room."
+			}
+		case "leave":
+			if sender == target {
+				if prevContent.Membership == "invite" {
+					return target + " rejected invite."
+				} else {
+					return target + " left the room."
+				}
+			} else if prevContent.Membership == "ban" {
+				return sender + " unbanned " + target + "."
+			} else if prevContent.Membership == "leave" {
+				return sender + " kicked " + target + "."
+			} else if prevContent.Membership == "invite" {
+				return sender + " withdrew " + target + "'s invite."
+			} else {
+				return target + " left the room"
+			}
+		}
+
+		return effect
+	},
+	"mRoomMessage": func(event *gomatrix.Event) interface{} {
+		switch event.Content["msgtype"] {
+		case "m.notice":
+			fallthrough
+		case "m.emote":
+			fallthrough
+		case "m.text":
+			fallthrough
+		default:
```

Comment:
I will be adding m.image etc, but that is a good point so I will combine all the fallthrough cases into one, just to explicitly show why they don't have a specific handler